main.py

```diff
@@ -0,0 +1,138 @@
+from fastapi import FastAPI, Depends, HTTPException
+from sqlalchemy.orm import Session
+
+import crud
+import models
+import schemas
+from database import SessionLocal
+
+app = FastAPI()
+
+
+def get_db() -> Session:
+    db = SessionLocal()
+
+    try:
+        yield db
+    finally:
+        db.close()
+
+
+def paginate(objects: list[any], skip: int, limit: int) -> list[any] | str:
+    amount_of_objects = len(objects)
+    if not 0 <= skip < amount_of_objects:
+        return (
+            "Skip should be greater or equal 0 and less then "
+            f"{amount_of_objects}"
+        )
+
+    if limit < 1:
+        return "Limit should be greater or equal 1"
+    return objects[skip : skip + limit]
+
+
+@app.post("/library/authors/", response_model=schemas.Author)
+def create_author(author: schemas.AuthorCreate, db: Session = Depends(get_db)):
+    if crud.get_author_by_name(db=db, name=author.name):
+        raise HTTPException(
+            status_code=400, detail="Such Author already exist"
+        )
+
+    return crud.create_author(db=db, author=author)
+
+
+@app.get("/library/authors/", response_model=list[schemas.Author])
+def read_authors(
+    skip: int = 0,
+    limit: int = 10,
+    db: Session = Depends(get_db),
+):
+    authors = crud.get_all_author(db=db)
+    page = paginate(authors, skip=skip, limit=limit)
```

Comment:
You're retrieving all items from db and then doing pagination on python side. It is not correct solution.
You can use .limit() and .offset() methods from SqlAlchemy to perform pagination on the DB side and not to retrieve redundant data from the db